src/TwilioProvider.php

```diff
@@ -22,19 +24,19 @@ public function boot()
                 );
             });
 
-        $this->app->bind(TwilioService::class, function () {
-            $config = $this->app['config']['services.twilio'];
-            $account_sid = Arr::get($config, 'account_sid');
-            $username = Arr::get($config, 'username');
-            if (! empty($username)) {
-                $password = Arr::get($config, 'password');
+        $this->app->bind(TwilioService::class, function (Application $app) {
```

Comment:
I can see no need in having `boot()` method in this service proveder.
All bindings can be moved into `register()`. Since all of them are specified as factory Closure, the particular components creation will be performed in lazy way.
Developer would be bale to override any of them any time, there is no need to wait for the booting stage.

Reply:
Looking at the docs, thinking that we might not be able to use a deferred provider here at all:
https://laravel.com/docs/7.x/providers#deferred-providers

> If your provider is only registering bindings in the service container, you may choose to defer its registration until one of the registered bindings is actually needed

We need to publish/merge config, so I'm not sure this will work.
I'll make a quick test app and see.

Reply:
Config merging will still work as service provider will be instantiating and invoked once its provided binding will be called. The only difference that this merging will be also lazy.

Publishing of the files will definitly work in case you specify provider class for the publish command:

```
php artisan vendor:publish --provider="NotificationChannels\Twilio\TwilioProvider"
```

In this case provider will be instantiating explicitly and its publishing will be performed.

Reply:
Fair point. Thanks, I'm not that familiar with deferred providers.